### PR TITLE
Add pdsClient method for adding authorizer

### PR DESCRIPTION
**Pull request by galxy25**
Add pdsClient method for authorizing a client to share records with other clients, integration tests.

### CODE REVIEW

File: pdsClient/pdsClient.go [hunk 1]
```diff
@@ -192,6 +236,76 @@ func (c *E3dbPDSClient) InternalGetRecord(ctx context.Context, recordID string)
...
+		UserID:     params.UserID,
+		ReaderID:   params.UserID,
+		RecordType: params.RecordType,
+	})
+	if err != nil {
+		return err
+	}
+	encryptedAccessKey := getAccessKeyResponse.EAK
+	if encryptedAccessKey == "" {
+		return errors.New(fmt.Sprintf("no access key exists for records of type %s", params.RecordType))
```
Comment (willmichael) [inline pdsClient/pdsClient.go]: Is this method a helper method for tests?

In the python SDK, if no EAK exists it is created and put for both the sharer and sharee to allow sharing of record types that don't yet exist

File: pdsClient/pdsClient.go [hunk 2]
```diff
@@ -192,6 +236,76 @@ func (c *E3dbPDSClient) InternalGetRecord(ctx context.Context, recordID string)
...
+	}
+	// TODO: Decrypt this key
+	// TODO: using the decrypted version of
+	// encryptedAccessKey and the public key of
+	// the reader specified in params,
+	// create an encrypted access key that
+	// the reader can decrypt
+	// Put the encrypted access key for the authorizer
```
Comment (willmichael) [inline pdsClient/pdsClient.go]: `// Put the encrypted access key for the reader`?

File: pdsClient/pdsClient.go (hunk 1 above)
Comment (galxy25) [reply pdsClient/pdsClient.go]: Yes, and tests have a MakeClientWriterForRecord that creates an access key for the sharer/writer of a record.

File: pdsClient/pdsClient.go (hunk 2 above)
Comment (galxy25) [reply pdsClient/pdsClient.go]: Good catch

**Pull request merged**